src/lib.rs

```diff
@@ -106,6 +108,40 @@ pub mod ext {
     pub use tantivy;
 }
 
+#[cfg(feature = "serde-bincode")]
+mod serialize {
+    pub fn serialize<T: ?Sized>(value: &T) -> bincode::Result<Vec<u8>>
+    where
+        T: serde::Serialize,
+    {
+        bincode::serialize(value)
+    }
```

Comment:
I'd suggest wrapping this in the crate `Error` so that you don't have to use feature flags in the `Store` methods. I.e., 

```rust
pub fn serialize<T: ?Sized>(value: &T) -> crate::err::Result<Vec<u8>>
    where
        T: serde::Serialize,
    {
        bincode::serialize(value).map_err(err::Error::Bincode)
    }
```

And do the same for `deserialize` as well as the `CBOR` ser/de methods. Then in for example `Store::create` you can do:

```rust
let serialized_inner = crate::serialize::serialize(inner)
    .map_err(sled::ConflictableTransactionError::Abort)?
```

with no feature flags.